File: modules/core-api/src/main/java/com/enonic/xp/schema/content/ContentTypesUpdatedEvent.java
package com.enonic.xp.schema.content;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

import com.enonic.xp.event.Event;

public class ContentTypesUpdatedEvent
    implements Event
{
    private List<ContentTypeName> names;


    private final Instant modifiedTime;

    public ContentTypesUpdatedEvent( final Instant modifiedTime )
    {
        names = new ArrayList<ContentTypeName>();
        this.modifiedTime = modifiedTime;
    }

    public void addContentTypeName( final ContentTypeName name ) {
        names.add( name );
    }

    public List<ContentTypeName> getNames()
    {
        return names;
    }

    public Instant getModifiedTime()
    {
        return modifiedTime;
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/schema/content/ContentTypesUpdatedEvent.ts
module api.schema.content {

    export class ContentTypesUpdatedEvent extends api.event.Event {

        private contentTypeNames: ContentTypeName[];

        private modifiedTime: Date;

        constructor(names: string[], modifiedTime: Date) {
            super();
            this.contentTypeNames = names.map((name: string) => {
                return new ContentTypeName(name);
            });
            this.modifiedTime = modifiedTime;
        }

        getContentTypeNames(): ContentTypeName[] {
            return this.contentTypeNames;
        }

        getModifiedTime(): Date {
            return this.modifiedTime;
        }

        static on(handler: (event: ContentTypesUpdatedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentTypesUpdatedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }


        static fromJson(json: ContentTypeUpdatedEventJson): ContentTypesUpdatedEvent {
            var modifiedTime = json.modifiedTime ? new Date(Date.parse(json.modifiedTime)) : null;
            return new ContentTypesUpdatedEvent(json.names, modifiedTime);
        }
    }

    export interface ContentTypeUpdatedEventJson {

        names: string[];

        modifiedTime: string;
    }
}